From 97454ce4fef91c08c7ee0a418f436cc2df6d8312 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:54 -0400
Subject: [fs] btrfs: remove dead code

Message-id: <1407179109-13714-48-git-send-email-sandeen@redhat.com>
Patchwork-id: 87030
O-Subject: [RHEL7.1 PATCH 047/422] btrfs: remove dead code
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Michal Nazarewicz <mina86@mina86.com>

commit 11850392ee1c600c7d40d93119daa72715bc959c
Author: Michal Nazarewicz <mina86@mina86.com>
Date:   Fri Nov 29 18:01:15 2013 +0100

    btrfs: remove dead code

    [commit 8185554d: fix incorrect inode acl reset] introduced a dead
    code by adding a condition which can never be true to an else
    branch.  The condition can never be true because it is already
    checked by a previous if statement which causes function to return.

    Signed-off-by: Michal Nazarewicz <mina86@mina86.com>
    Reviewed-By: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/acl.c b/fs/btrfs/acl.c
index 0890c83..460f36b 100644
--- a/fs/btrfs/acl.c
+++ b/fs/btrfs/acl.c
@@ -225,13 +225,8 @@ int btrfs_init_acl(struct btrfs_trans_handle *trans,
   ret = posix_acl_create(&acl, GFP_NOFS, &inode->i_mode);
   if (ret < 0)
    return ret;
-
-  if (ret > 0) {
-   /* we need an acl */
+  if (ret > 0) /* we need an acl */
    ret = btrfs_set_acl(trans, inode, acl, ACL_TYPE_ACCESS);
-  } else if (ret < 0) {
-   cache_no_acl(inode);
-  }
  } else {
   cache_no_acl(inode);
  }
-- 
1.7.1